content/mocking-mistakes.md

```diff
@@ -0,0 +1,85 @@
+
+## Mitigating mistaken mocks
+
+Having learned about this mistake, how do we mitigate it? Fortunately, it's easy to replace `mock.assert_called_once()` with `assert mock.call_count == 1` - a statement that will only return true if the mock has actually been called once. Similarly, `mock.assert_not_called()` can be replaced with `assert mock.call_count == 0`. This solution was highlighted in [this post](http://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html) on the Yelp engineering blog. Since learning about it, I've started incorporating this style of assertion into all cases where I want to check how often something was called, but don't care about what arguments it was called with.
+
+An alternative is to use the [`autospec` property](https://docs.python.org/3/library/unittest.mock.html#autospeccing) of the mock object. From the `mock` documentation:
+
+> Auto-speccing creates mock objects that have the same attributes and methods as the objects they are replacing, and any functions and methods (including constructors) have the same call signature as the real object. This ensures that your mocks will fail in the same way as your production code if they are used incorrectly.
+
+When we use the `autospec=True` argument in our `@mock.patch` decorator, our mock object will only exhibit the methods that actually exist on the original object we are replacing.
+
+```python
+class ClientTestCase(unittest.TestCase)
+    @mock.patch(self, 'client.MyAPIClient._create_headers', autospec=True)
+    def test_client_get(mock_create_headers):
+
+        response = client.MyAPIClient.get(url="http://engineroom.trackmaven.com/")
+
+        mock_create_headers.assert_called_once()
+```
+
+This means that `assert_called_once` and `assert_not_called` are exposed as the deceptive fake methods that they were all along:
+
+```
+$ python tests.py
+...E
+======================================================================
+ERROR: test_client_get (__main__.ClientTestCase)
+----------------------------------------------------------------------
+Traceback (most recent call last):
+  File "/home/jonathan/.virtualenvs/blog_posts/local/lib/python2.7/site-packages/mock.py", line 1201, in patched
+    return func(*args, **keywargs)
+  File "tests.py", line 47, in test_client_get
+    mock_create_handlers.assert_called_once()
+  File "/home/jonathan/.virtualenvs/blog_posts/local/lib/python2.7/site-packages/mock.py", line 658, in __getattr__
+    raise AttributeError("Mock object has no attribute %r" % name)
+AttributeError: Mock object has no attribute 'assert_called_once'
+
+----------------------------------------------------------------------
+Ran 4 tests in 0.004s
+
+FAILED (errors=1)
+```
+
+Autospeccing is useful for more things than preventing cheeky errors, so definitely take a look at it.
```

Comment:
Does this remove the ability to call `mock_create_headers.call_count`? If so, maybe a note mentioning it one way or the other...